### PR TITLE
Remove network detection

### PR DESCRIPTION
**Pull request by dcousens**
See https://github.com/bitcoinjs/bitcoinjs-lib/issues/402 for discussion.

### CODE REVIEW

**Comment by dcousens** [comment]
This does not detection entirely, yet. 

> In case of detection of a network, require a list of possible networks passed in.

Any thoughts on how this should work with `ECPair.fromWIF`?
What would be **the** best succinct syntax for that?

File: src/ecpair.js [hunk 1]
```diff
@@ -88,7 +78,21 @@ ECPair.fromWIF = function (string) {
...
   }
 
-  var network = findNetworkByWIFVersion(version)
+  var network
+
+  // list of networks?
+  if (Array.isArray(networks)) {
+    network = networks.filter(function (network) {
+      return network.wif === version
+    }).pop() || {}
```
Comment (dcousens) [inline src/ecpair.js]: @weilu this could probably be more efficient,  but,  it'll do for now

File: src/ecpair.js [hunk 2]
```diff
@@ -67,7 +57,7 @@ ECPair.fromPublicKeyBuffer = function (buffer, network) {
   })
 }
 
-ECPair.fromWIF = function (string) {
+ECPair.fromWIF = function (string, networks) {
```
Comment (dcousens) [inline src/ecpair.js]: ping @weilu

File: src/transaction_builder.js [hunk 3]
```diff
@@ -188,7 +190,7 @@ TransactionBuilder.prototype.addOutput = function (scriptPubKey, value) {
 
   // Attempt to get a valid address if it's a base58 address string
   if (typeof scriptPubKey === 'string') {
-    scriptPubKey = Address.toOutputScript(scriptPubKey)
+    scriptPubKey = Address.toOutputScript(scriptPubKey, this.network)
```
Comment (weilu) [inline src/transaction_builder.js]: There's a potential issue with this use: `var cb = TransactionBuilder.fromTransaction` always assumes bitcoin as the network. If that's called, followed by a call to `cb.addOutput([a non-bitcoin address], value)` without manually setting the network on `cb` it's no good.

**Comment by weilu** [comment]
integration tests are passing locally. Not sure what's wrong with travis =/

File: src/transaction_builder.js (hunk 3 above)
Comment (dcousens) [inline src/transaction_builder.js]: Good point,  `fromTransaction` will need a `network` parameter also.

**Comment by dcousens** [comment]
Rebased and cherry picked 299cfce1f4b2c5af149a468714421040876d9de5 on top.

**Comment by dcousens** [comment]
@weilu I addressed the above issues,  looking to keep moving forward on this.   Happy for any post reviews.

**Comment by weilu** [comment]
LGTM

**Pull request merged**